py-polars/tests/unit/interchange/test_roundtrip.py

```diff
@@ -44,42 +49,170 @@ def test_roundtrip_pyarrow_parametric(df: pl.DataFrame) -> None:
     assert_frame_equal(result, df, categorical_as_str=True)
 
 
-@given(dataframes(allowed_dtypes=protocol_dtypes))
 @pytest.mark.filterwarnings(
     "ignore:.*PEP3118 format string that does not match its itemsize:RuntimeWarning"
 )
-def test_roundtrip_pandas_parametric(df: pl.DataFrame) -> None:
+@given(dataframes(allowed_dtypes=protocol_dtypes))
+def test_to_dataframe_pandas_parametric(df: pl.DataFrame) -> None:
     dfi = df.__dataframe__()
     df_pd = pd.api.interchange.from_dataframe(dfi)
     result = pl.from_pandas(df_pd, nan_to_null=False)
     assert_frame_equal(result, df, categorical_as_str=True)
 
 
+@pytest.mark.filterwarnings(
+    "ignore:.*PEP3118 format string that does not match its itemsize:RuntimeWarning"
+)
 @given(
     dataframes(
         allowed_dtypes=protocol_dtypes,
         excluded_dtypes=[pl.Categorical],
         chunked=False,
     )
 )
-@pytest.mark.filterwarnings(
-    "ignore:.*PEP3118 format string that does not match its itemsize:RuntimeWarning"
-)
-def test_roundtrip_pandas_zero_copy_parametric(df: pl.DataFrame) -> None:
+def test_to_dataframe_pandas_zero_copy_parametric(df: pl.DataFrame) -> None:
     dfi = df.__dataframe__(allow_copy=False)
     df_pd = pd.api.interchange.from_dataframe(dfi, allow_copy=False)
     result = pl.from_pandas(df_pd, nan_to_null=False)
     assert_frame_equal(result, df, categorical_as_str=True)
 
 
-def test_roundtrip_pandas_boolean_subchunks() -> None:
+@given(
+    dataframes(
+        allowed_dtypes=protocol_dtypes,
+        excluded_dtypes=[
+            pl.Categorical,  # Categoricals read back as Enum types
+        ],
+    )
+)
+def test_from_dataframe_pyarrow_parametric(df: pl.DataFrame) -> None:
+    df_pa = df.to_arrow()
+    result = pl.from_dataframe(df_pa)
+    assert_frame_equal(result, df, categorical_as_str=True)
+
+
+@given(
+    dataframes(
+        allowed_dtypes=protocol_dtypes,
+        excluded_dtypes=[
+            pl.Categorical,  # Polars copies the categories to construct a mapping
+            pl.Boolean,  # pyarrow exports boolean buffers as byte-packed: https://github.com/apache/arrow/issues/37991
+        ],
+        chunked=False,
+    )
+)
+def test_from_dataframe_pyarrow_zero_copy_parametric(df: pl.DataFrame) -> None:
+    df_pa = df.to_arrow()
+    result = pl.from_dataframe(df_pa, allow_copy=False)
+    assert_frame_equal(result, df)
+
+
+@given(
+    dataframes(
+        allowed_dtypes=protocol_dtypes,
+        excluded_dtypes=[
+            pl.Categorical,  # Categoricals come back as Enums
+            # large string not yet supported by pandas
+            # https://github.com/pandas-dev/pandas/issues/56702
+            pl.String,
+            pl.Float32,  # NaN values come back as nulls
+            pl.Float64,  # NaN values come back as nulls
+            # pandas exports nanosecond pyarrow types incorrectly
+            # https://github.com/pandas-dev/pandas/issues/56712
+            pl.Datetime("ns"),
```

Comment:
note to self, this can be removed after the next pandas release

Reply:
I'll go back and update the tests after pandas releases (remove the xfail, update some of the parametrics).